File: app13_ruby_FastApi/backend/db/migrations/008_create_users.rb
# frozen_string_literal: true

Sequel.migration do
  change do
    create_table(:users) do
      primary_key :id
      String :username, null: false, unique: true
      String :password_digest, null: false
      String :role, null: false, default: "ADMIN"

      constraint(:users_role_check) { Sequel.lit("role IN ('ADMIN')") }
    end
  end
end

File: app14_LUA_UNITY/backend/db/migrations/008_create_users.lua
local schema = require("lapis.db.schema")
local types = schema.types
local db = require("lapis.db")

return {
  up = function()
    schema.create_table("users", {
      { "id", types.serial({ primary_key = true }) },
      { "username", types.varchar({ null = false }) },
      { "password_digest", types.varchar({ null = false }) },
      { "role", types.varchar({ null = false, default = "ADMIN" }) }
    })

    db.query("create unique index users_username_idx on users (username)")
    db.query("alter table users add constraint users_role_check check (role in ('ADMIN'))")
  end,
  down = function()
    schema.drop_table("users")
  end
}
